### DISCUSSION TITLE
PuLP solvers do not respond to options fed to them

### DISCUSSION THREAD

**Discussion by JorritHimself** [coin-or/pulp#402] [General]
Crossposted on stackoverflow: https://stackoverflow.com/questions/66114805/pulp-solvers-do-not-respond-to-options-fed-to-them

So I've got a fairly large optimization problem and I'm trying to solve it within a sensible amount of time. Ive set it up as:

    import pulp as pl
    my_problem = LpProblem("My problem",LpMinimize)`
    # write to problem file
    my_problem.writeLP("MyProblem.lp")

And then alternatively
```
solver = CPLEX_CMD(timeLimit=1, gapRel=0.1)
status = my_problem .solve(solver)
```
```
solver = pl.apis.CPLEX_CMD(timeLimit=1, gapRel=0.1)
status = my_problem .solve(solver)
```
```
path_to_cplex = r'C:\Program Files\IBM\ILOG\CPLEX_Studio1210\cplex\bin\x64_win64\cplex.exe' # and yes this is the actual 
path on my machine
solver = pl.apis.cplex_api.CPLEX_CMD(timeLimit=1, gapRel=0.1, path=path_to_cplex)
status = my_problem .solve(solver)
```
```
solver = pl.apis.cplex_api.CPLEX_CMD(timeLimit=1, gapRel=0.1, path=path_to_cplex)
status = my_problem .solve(solver)
```

It runs in each case. However, the solver does not repond to the timeLimit or gapRel instructions. If I use timelimit it does warn this is depreciated for timeLimit. Same for fracgap: it tells me I should use relGap. So somehow I am talking to the solver. However, nor matter what values i pick for timeLimit and relGap, it always returns the exact same answer and takes the exact same amount of time (several minutes). Also, I have tried alternative solvers, and I cannot get any one of them to accept their variants of time limits or optimization gaps. In each case, the problem solves and returns an status: optimal message. But it just ignores the time limit and gap instructions.

Any ideas?

**Comment by AlexFleischerParis** [comment]
I answered at https://stackoverflow.com/questions/66114805/pulp-solvers-do-not-respond-to-options-fed-to-them

**Comment by pchtsp** [comment]
Thanks @AlexFleischerParis. I'm still curious on why the arguments do not work. Could you share your problem data in python code, mps file or json file? more info on how to export a model: https://coin-or.github.io/pulp/guides/how_to_export_models.html

**Comment by JorritHimself** [comment]
Thanks. I had seen that variation but figured it was from a much earlier version of PuLP. 
Am experimenting with small and bigger problems and will report back on what I find.

**Comment by beevabeeva** [comment]
I was experiencing the same/similar issue.
That is how I found the Google groups and then this Discussions site. It is really awesome that Franco and Stuart make themselves available to help the pulp community, and doing so with patience.

It turned out that my (massive 2GB + ) pulp model was taking in dummy data I'd created, that made the problem infeasible/malformed - and correctly so.

I had to keep manually killing the process (sometimes after letting it run for 20 hours with no solution/feedback). I tried running with a smaller input size. This did finish running after about 40-50 seconds, but the `timeLimit` parameter was still being ignored.

After a good night's sleep, I realised my problem with the data. But after an additional good night's sleep, I realised I should have run my toy model (which I knew was correct) with the `timeLimit` parameter. 

It is interesting that this behaviour occurs though. I hope this reply can aid in pointing to the root cause.

**Reply by pchtsp** [reply]
Thanks for the comment. Sadly, it does not help in pointing the root cause. What would help is to have an example (python code, mps model, json model) we can run in our machines that reproduces the issue. More information on how to do that [here](https://coin-or.github.io/pulp/guides/how_to_debug.html#giving-feedback-and-asking-for-help).
